service/src/main/java/gov/nasa/pds/api/registry/model/RequestAndResponseContext.java

```diff
@@ -143,8 +143,9 @@ private RequestAndResponseContext(
 	public final GroupConstraint getPresetCriteria() { return this.presetCriteria; };
 	public ProductVersionSelector getSelector() { return this.selector; }
 
-	public boolean isSingular() { return this.getStart() == -1 && this.isSummaryOnly(); } // isSummaryOnly implies original limit value == 0
-	public boolean isSummaryOnly() { return summaryOnly; }
+
+	public boolean isSingular() { return this.returnSingularDatum; }
```

Comment:
If I am not mistaken, I mean if I understand what you do with returnSingularDatum: we don't want to return a specific format when limit=0 or 1. The format singular or plural is defined by the end-point.

It is very likely that I don't understand what you do here, so we can discuss that later.

Reply:
Previously, there was some slightly-murky (to someone who isn't already familiar with it) state checking of (values pulled from) the `URIParameters` going on to determine whether the API route was a return-element or return-collection type.

That implicit logic has been made explicit and encapsulated within the `URIParameters` object.

The logic itself has not changed meaningfully - if the setters for `start` or `limit` are called on `URIParameters` object, the flag changes from return-element (returnSingularDatum=True) to return-collection(returnSingularDatum=False).  I didn't want to overcomplicate the changes, but if this is sufficiently unclear, a potential improvement would be to change the relevant member from `boolean isSingle` to an enumerated type `routeReturnType` with values `SingleElement` and `ElementCollection` or something similar.

Reply:
@tloubrieu-jpl unless I'm missing something, this is resolved - feel free to re-open if you have outstanding questions